tutorials/metrics.md

```diff
@@ -36,14 +36,64 @@ Beside the [default golang based metrics][2], CRI-O provides the following addit
+  `UpdateContainerResources`, `UpdateRuntimeConfig`, `Version`
+
+- Available error categories for `crio_image_pulls_failures`:
+  - `UNKNOWN`: The default label which gets applied if the error is not known
+  - `CONNECTION_REFUSED`: The local network is down or the registry refused the
+    connection
+  - `CONNECTION_TIMEOUT`: The connection timed out during the image download
+  - `NOT_FOUND`: The registry does not exist at the specified resource
+  - `BLOB_UNKNOWN`: This error may be returned when a blob is unknown to the
+    registry in a specified repository. This can be returned with a standard get
+    or if a manifest references an unknown layer during upload.
+  - `BLOB_UPLOAD_INVALID`: The blob upload encountered an error and can no
+    longer proceed.
+  - `BLOB_UPLOAD_UNKNOWN`: If a blob upload has been cancelled or was never
+    started, this error code may be returned.
+  - `DENIED`: The access controller denied access for the operation on a
+    resource.
+  - `DIGEST_INVALID`: When a blob is uploaded, the registry will check that the
+    content matches the digest provided by the client. The error may include a
+    detail structure with the key "digest", including the invalid digest string.
+    This error may also be returned when a manifest includes an invalid layer
+    digest.
+  - `MANIFEST_BLOB_UNKNOWN`: This error may be returned when a manifest blob is
+    unknown to the registry.
+  - `MANIFEST_INVALID`: During upload, manifests undergo several checks ensuring
+    validity. If those checks fail, this error may be returned, unless a more
+    specific error is included. The detail will contain information the failed
+    validation.
+  - `MANIFEST_UNKNOWN`: This error is returned when the manifest, identified by
+    name and tag is unknown to the repository.
+  - `MANIFEST_UNVERIFIED`: During manifest upload, if the manifest fails
+    signature verification, this error will be returned.
+  - `NAME_INVALID`: Invalid repository name encountered either during manifest
+    validation or any API operation.
+  - `NAME_UNKNOWN`: This is returned if the name used during an operation is
+    unknown to the registry.
+  - `SIZE_INVALID`: When a layer is uploaded, the provided size will be checked
+    against the uploaded content. If they do not match, this error will be
+    returned.
+  - `TAG_INVALID`: During a manifest upload, if the tag in the manifest does not
+    match the uri tag, this error will be returned.
+  - `TOOMANYREQUESTS`: Returned when a client attempts to contact a service too
+    many times
+  - `UNAUTHORIZED`: The access controller was unable to authenticate the client.
+    Often this will be accompanied by a Www-Authenticate HTTP response header
+    indicating how to authenticate.
+  - `UNAVAILABLE`: Returned when a service is not available
+  - `UNSUPPORTED`: The operation was unsupported due to a missing implementation
+    or invalid set of parameters.
```

Comment:
About a third of the above lines are missing ending periods.  I think they all should have one.

Reply:
Makes sense, fixed that :+1: